src/artrace.py

```diff
@@ -804,59 +805,116 @@ def refine_traces(binarr, outpar, extrap_cent, extrap_diff, extord, orders, disp
     return extfit, outpar
 
 
-def model_tilt_test(slf, det, msarc, guesstilts=None, censpec=None, plotQA=False, maskval=-999999.9):
+def model_tilt(slf, det, msarc, censpec=None, maskval=-999999.9):
     """
     This function performs a PCA analysis on the arc tilts for a single spectrum (or order)
     """
-    msgs.work("Detecting lines..")
-    tampl, tcent, twid, w, satsnd, detsub = ararc.detect_lines(slf, det, msarc, censpec=censpec)
+    msgs.work("Detecting lines")
+    tampl, tcent, twid, w, satsnd, _ = ararc.detect_lines(slf, det, msarc, censpec=censpec)
     satval = slf._spect['det'][det-1]['saturation']*slf._spect['det'][det-1]['nonlinear']
-    # Order of the polynomials to be used when fitting the tilts with a 2D polynomial
-    fitxy = [slf._argflag['trace']['orders']['tiltorder'], 8]
+    if slf._argflag['trace']['orders']['tilts'].lower() == "zero":
+        # Assuming there is no spectral tilt
+        tilts = np.outer(np.linspace(0.0, 1.0, msarc.shape[0]), np.ones(msarc.shape[1]))
+        return tilts, satsnd
+    # Order of the polynomials to be used when fitting the tilts.
+    fitxy = [slf._argflag['trace']['orders']['tiltorder'], 1]
     arcdet = (tcent[w]+0.5).astype(np.int)
-    # Start by masking every row, then later unmask the rows with usable arc lines
-    maskrows = np.ones(msarc.shape[0], dtype=np.int)
-    totnum = 0
-    if np.size(w[0]) > totnum:
-        totnum = np.size(w[0])
-
-    ordcen = slf.GetFrame(slf._pixcen, det)
-    # Trace the tilts
-    plt.clf()
-    if slf._argflag['trace']['orders']['tilts'] == 'fit1D':
-        # Go along each order and fit the tilts in 1D
-        tiltang = -999999.9*np.ones(arcdet.size)
-        centval = -999999.9*np.ones(arcdet.size)
+    ampl = tampl[w]
+
+    # Determine the best lines to use to trace the tilts
+    trthrsh = 1000.0
+    ncont = 15
+    aduse = np.zeros(arcdet.size, dtype=np.bool)  # Which lines should be used to trace the tilts
+    w = np.where(ampl >= trthrsh)
+    aduse[w] = 1
+    # Remove lines that are within ncont pixels
+    nuse = np.sum(aduse)
+    detuse = arcdet[aduse]
+    idxuse = np.arange(arcdet.size)[aduse]
+    olduse = aduse.copy()
+    for s in xrange(nuse):
+        w = np.where((np.abs(arcdet-detuse[s]) <= ncont) & (np.abs(arcdet-detuse[s]) >= 1.0))[0]
+        for u in xrange(w.size):
+            if ampl[w[u]] > ampl[olduse][s]:
+                aduse[idxuse[s]] = 0
+                break
+
```

Comment:
It's searches the neighborhood of each detected line to see if there are other detected lines within ncont pixels. If there is, it will only take the brightest line. I'll need to think about this some more...